File: src/main/java/info/skyblond/archivedag/controller/http/AuthController.java
package info.skyblond.archivedag.controller.http;

import info.skyblond.archivedag.model.bo.JWTAuthResponse;
import info.skyblond.archivedag.model.bo.JwtAuthRequest;
import info.skyblond.archivedag.security.JwtTokenManager;
import lombok.extern.slf4j.Slf4j;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Slf4j
@RestController
@RequestMapping("/public/auth")
public class AuthController {
    private final JwtTokenManager jwtTokenManager;
    private final AuthenticationManager authenticationManager;
    private final UserDetailsService userDetailsService;

    public AuthController(JwtTokenManager jwtTokenManager, AuthenticationManager authenticationManager, UserDetailsService userDetailsService) {
        this.jwtTokenManager = jwtTokenManager;
        this.authenticationManager = authenticationManager;
        this.userDetailsService = userDetailsService;
    }

    @PostMapping
    public JWTAuthResponse signJwtToken(
            @RequestBody JwtAuthRequest request
    ) {
        this.authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(
                request.getUsername(), request.getPassword()
        ));
        UserDetails userDetails = this.userDetailsService.loadUserByUsername(request.getUsername());
        String token = this.jwtTokenManager.generateToken(userDetails);
        return new JWTAuthResponse(token);
    }
}

File: backend/src/main/kotlin/info/skyblond/archivedag/arudaz/controller/http/AuthController.kt
package info.skyblond.archivedag.arudaz.controller.http

import info.skyblond.archivedag.arudaz.model.JWTAuthResponse
import info.skyblond.archivedag.arudaz.model.JwtAuthRequest
import info.skyblond.archivedag.arudaz.service.JwtTokenService
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/public/auth")
class AuthController(
    private val jwtTokenService: JwtTokenService,
    private val authenticationManager: AuthenticationManager,
    private val userDetailsService: UserDetailsService
) {
    @PostMapping
    fun signJwtToken(
        @RequestBody request: JwtAuthRequest
    ): JWTAuthResponse {
        authenticationManager.authenticate(
            UsernamePasswordAuthenticationToken(
                request.username, request.password
            )
        )
        val userDetails = userDetailsService.loadUserByUsername(request.username)
        val token = jwtTokenService.generateToken(userDetails)
        return JWTAuthResponse(token)
    }
}
